modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/admin/throttlingpolicy/ThrottlePolicyExportImportTestCase.java

```diff
@@ -0,0 +1,597 @@
+        //Assert the status code and policy ID
+        Assert.assertEquals(addedPolicy.getStatusCode(), HttpStatus.SC_CREATED);
+        ApplicationThrottlePolicyDTO addedPolicyDTO = addedPolicy.getData();
+        applicationPolicyId = addedPolicyDTO.getPolicyId();
+        Assert.assertNotNull(applicationPolicyId, "The policy ID cannot be null or empty");
+        ApplicationPolicyDTO.setPolicyId(applicationPolicyId);
+        ApplicationPolicyDTO.setIsDeployed(true);
+        //Verify the created application throttling policy DTO
+        adminApiTestHelper.verifyApplicationThrottlePolicyDTO(ApplicationPolicyDTO, addedPolicyDTO);
+    }
+
+    /**
+     * Adds a new custom policy
+     */
+    public void addCustomPolicy() {
+        //Create the custom throttling policy DTO
+        String description = "This is a test custom throttle policy";
+        String siddhiQuery = "FROM RequestStream\nSELECT userId, ( userId == 'admin@carbon.super' ) AS isEligible, "
+                + "str:concat('admin@carbon.super','') as throttleKey\nINSERT INTO EligibilityStream; \n\nFROM "
+                + "EligibilityStream[isEligible==true]#throttler:timeBatch(1 min) \nSELECT throttleKey, (count(userId) >= 10) "
+                + "as isThrottled, expiryTimeStamp group by throttleKey \nINSERT ALL EVENTS into ResultStream;";
+        String keyTemplate = "$userId";
+        CustomPolicyDTO = DtoFactory.createCustomThrottlePolicyDTO(customPolicyName, description, false, siddhiQuery,
+                keyTemplate);
+
+        ApiResponse<CustomRuleDTO> addedPolicy;
+        try {
+            //Add the custom throttling policy
+            addedPolicy = restAPIAdmin.addCustomThrottlingPolicy(CustomPolicyDTO);
+
+            //Assert the status code and policy ID
+            Assert.assertEquals(addedPolicy.getStatusCode(), HttpStatus.SC_CREATED);
+            CustomRuleDTO addedPolicyDTO = addedPolicy.getData();
+            customPolicyId = addedPolicyDTO.getPolicyId();
+            Assert.assertNotNull(customPolicyId, "The policy ID cannot be null or empty");
+            CustomPolicyDTO.setPolicyId(customPolicyId);
+            CustomPolicyDTO.setIsDeployed(true);
+            //Verify the created custom throttling policy DTO
+            adminApiTestHelper.verifyCustomThrottlePolicyDTO(CustomPolicyDTO, addedPolicyDTO);
+        } catch (ApiException e) {
+            if (userMode == TestUserMode.TENANT_ADMIN) {
+                Assert.assertEquals(e.getCode(), HttpStatus.SC_FORBIDDEN);
+            }
+        }
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Exported Sample Custom ThrottlePolicy" )
+    public void testCustomThrottlePolicyExport() throws Exception {
+        if (userMode == TestUserMode.TENANT_ADMIN) {
+            throw new SkipException("Skipping the export custom policy");
+        }
+        exportedFileCustomPolicy = exportArtifact(user.getUserName(), user.getPassword(), customPolicyName,
```

Comment:
Can use the generated API class methods (in org.wso2.am.integration.clients.admin.api package) for these export/import operations. No need to write http call generation methods from the scratch.

Reply:
Fixed with 3106d801961d2d8e82a5a35117427b8337a92455